Sources/GtkBackend/GtkBackend.swift

```diff
@@ -1581,3 +1612,169 @@ extension UnsafeMutablePointer {
+                insert(child: minuteSecondSeparator!, after: minutePicker)
+                insert(child: secondPicker!, after: minuteSecondSeparator!)
+            }
+        } else {
+            if let minuteSecondSeparator {
+                remove(minuteSecondSeparator)
+                self.minuteSecondSeparator = nil
+            }
+            if let secondPicker {
+                remove(secondPicker)
+                self.secondPicker = nil
+            }
+        }
+
+        let minutesRange = calendar.range(of: .minute, in: .hour, for: date) ?? 0..<60
+        minutePicker.setRange(
+            min: Double(minutesRange.lowerBound),
+            max: Double(minutesRange.upperBound - 1)
+        )
+        minutePicker.text = "\(components.minute!)"
+        minutePicker.valueChanged = { [unowned self] minutePicker in
+            guard let value = Int(exactly: minutePicker.value),
+                let newDate = calendar.date(bySetting: .minute, value: value, of: date)
+            else {
+                return
+            }
+            self.onChange?(newDate)
+        }
+
+        let hoursRange = calendar.range(of: .hour, in: .day, for: date)
+        self.hourCycle = (calendar.locale ?? .current).hourCycle
+        let effectiveHours = hoursRange?.map {
+            TimePicker.transformToRange($0, hourCycle: self.hourCycle)
+        }
+
+        hourPicker.setRange(
+            min: effectiveHours?.min().map(Double.init(_:))
+                ?? TimePicker.minHour(for: self.hourCycle),
+            max: effectiveHours?.max().map(Double.init(_:))
+                ?? TimePicker.maxHour(for: self.hourCycle)
+        )
+
+        if self.hourCycle == .oneToTwelve || self.hourCycle == .zeroToEleven {
+            if let amPmPicker {
+                // update strings if necessary
+            } else {
+                amPmPicker = DropDown(strings: [calendar.amSymbol, calendar.pmSymbol])
+                add(amPmPicker!)
+            }
+        } else {
+            if let amPmPicker {
+                remove(amPmPicker)
+                self.amPmPicker = nil
+            }
+        }
+
+        hourPicker.text =
+            "\(TimePicker.transformToRange(components.hour!, hourCycle: self.hourCycle))"
+        hourPicker.valueChanged = { [unowned self] hourPicker in
+            guard let value = Int(exactly: hourPicker.value),
+                let newDate = calendar.date(bySetting: .hour, value: value, of: date)
+            else {
+                return
+            }
+            self.onChange?(newDate)
+        }
+    }
+
+    private static func transformToRange(_ value: Int, hourCycle: Locale.HourCycle) -> Int {
+        switch hourCycle {
+            case .zeroToEleven: value % 12
+            case .oneToTwelve: (value + 11) % 12 + 1
+            case .zeroToTwentyThree: value % 24
+            case .oneToTwentyFour: (value + 23) % 24 + 1
+            #if os(macOS)
+                @unknown default: fatalError()
```

Comment:
Add a custom error message

Reply:
This entire class is not really complete, but I can do that for the `fatalError`s I guess.